lib/iterator.js

```diff
@@ -199,6 +199,68 @@ class Iterator {
     return result + suffix;
   }
 
+  // Find value in this iterator compared by `comparator` on values
+  // obtained by applying `accessor` to each value.
+  // Signature: comparator, [accessor[, thisArg]]
+  //   comparator <Function> returns true if new value should be accepted.
+  //     currValue <any> current value, starts with undefined
+  //     nextValue <any> next value
+  //     Returns: <boolean> true if next value should be accepted.
+  //   accessor <Function> gets value to compare by, current iterator value
+  //       is used by default.
+  //     value <any> current iterator value.
+  //     Returns: <any> value to compare by.
+  //   thisArg <any> value to be used as `this` when calling `accessor` and
+  //       `comparator`.
+  // Returns: last iterator value where `comparator` returned `true` or
+  //     `undefined` if iterator is empty.
```

Comment:
It's not necessarily `undefined` only in cases when the iterator is empty, it will also be `undefined` if `comparator` didn't return `true` for any of the values.